internal/provider/resource_vault_cluster.go

```diff
@@ -1212,6 +1275,25 @@ func getValidObservabilityConfig(config map[string]interface{}) (*vaultmodels.Ha
 		}
 	}
 
+	if elasticsearchEndpoint != "" || elasticsearchDataset != "" || elasticsearchUser != "" || elasticsearchPassword != "" {
+		if observabilityConfig != nil {
+			return nil, tooManyProvidersErr
+		}
+
+		if elasticsearchEndpoint == "" || elasticsearchDataset == "" || elasticsearchUser == "" || elasticsearchPassword == "" {
+			missingParamErr = diag.Errorf("elasticsearch configuration is invalid: configuration information missing")
+		}
+
+		observabilityConfig = &vaultmodels.HashicorpCloudVault20201125ObservabilityConfig{
+			Elasticsearch: &vaultmodels.HashicorpCloudVault20201125Elasticsearch{
+				Endpoint: elasticsearchEndpoint,
+				User:     elasticsearchUser,
+				Password: elasticsearchPassword,
+				Dataset:  elasticsearchDataset,
```

Comment:
doesn't this break updates? I'd ask you to manually test the creation of a cluster with elasticserach configured as an observability provider for merics and audit logs, then update the settings and apply the terraform changes. That would allow you to validate that the Create and Update methods work fine with the new provider, as well as fetching info from the server and transforming it to the TF representation

Reply:
How can I test this update? I attempted to use the `hcloud hcp dev run` command in the root directory of this project to see if I can push the updates to my PRDE, but I get an error stating that I cannot use this command because it requires that the repository has a makefile in its root directory.

Reply:
a TF provider like this one for HCP is not a control-plane service that you'd run in PRDE, it's a module for Terraform that our customers import in their `.tf` files.

To test your changes you'd have to use the Terraform module like a customer would, but consuming the locally built provider instead of allowing Terraform to fetch from the Terraform registry. For instructions on how to build this provider see https://github.com/hashicorp/terraform-provider-hcp/tree/main/contributing#building-the-provider

For instructions on how to use the `vault_cluster` resource see https://registry.terraform.io/providers/hashicorp/hcp/latest/docs/resources/vault_cluster. There are also instructions for authenticating https://registry.terraform.io/providers/hashicorp/hcp/latest/docs/guides/auth

Also, I didn't have success trying to get Terraform to target anything other than the prod env, so I'd suggest to just create the resources in prod

Feel free to reach out if it's still unclear how you can manually verify your changes

Reply:
I finished writing the terraform script and it's included here for convienience.

```
terraform {
  required_providers {
    hcp = {
      source  = "localhost/providers/hcp"
      version = "0.0.1"
    }
  }
}

provider "hcp" {
  client_id = "fBBIue3fpg7nrV44b6Ej7rCAvSyPKIQR"
  client_secret = "dToqp6oq5CtmWwOoz78SfdISG95fYgvKppt4jXIi_66ekeJWMcfG4QR2fIq6Mu7B"
  project_id = "1fd2e0dc-6243-4468-9999-13576fe2c548"
}

provider "aws" {
  region="us-west-2"
}

resource "hcp_hvn" "example" {
  hvn_id         = "awshvn"
  cloud_provider = "aws"
  region         = "us-west-2"
  cidr_block     = "172.25.16.0/20"
}

resource "hcp_vault_cluster" "example" {
  cluster_id = "vault-cluster"
  hvn_id     = hcp_hvn.example.hvn_id
  tier = "standard_small"
  audit_log_config {
    elasticsearch_user="user"
    elasticsearch_password="password"
    elasticsearch_endpoint="https://elasticsearch"
  }
}
```

I get the following error that I believe is expected because elasticsearch may not be fully implemented as an observability provider as of now?

`Error: elasticsearch configuration is invalid: configuration information missing`.

Reply:
I mistakenly added `elasticsearch_dataset` as a user-validated field. After updating, this works as intended.

Reply:
so my concern here wasn't even the issue you hit and fixed, but instead that by setting `Dataset: elasticsearchDataset` you would cause the Dataset field to be set in an update request, which should cause the server to reject the request because that field cannot be set by the caller. While I think that the right approach would be to not set the field at all if it's working then it's ok